### PR TITLE
Molecule Measurements

### PR DESCRIPTION
**Pull request by dgasmith**
Adds several routines for computing distances, angles, and dihedrals within QCElemental. In addition adds a `Molecule.measure` command to check internal values.

### CODE REVIEW

File: qcelemental/util/misc.py [hunk 1]
```diff
@@ -130,3 +130,173 @@ def unnp(dicary, flat=False, _path=None):
...
     return ndicary
+
+
+def _norm(points):
+    """
+    Return the Frobenius norm across axis=-1, NumPy's internal norm is crazy slow (~4x)
+    """
+
+    tmp = np.atleast_2d(points)
+    return np.sqrt(np.einsum("ij,ij->i", tmp, tmp))
```
Comment (loriab) [inline qcelemental/util/misc.py]: not duplicating https://github.com/MolSSI/QCElemental/blob/81c65d815fbb18b8bb689e4a1c4c2b2c896d5c56/qcelemental/util/misc.py#L9-L18, is it?

File: qcelemental/util/misc.py [hunk 2]
```diff
@@ -130,3 +130,173 @@ def unnp(dicary, flat=False, _path=None):
...
+
+def measure_coordinates(coordinates, measurements, degrees=False):
+    """
+    Measures a geometry array based on indices provided, automatically detects distance, angle,
```
Comment (loriab) [inline qcelemental/util/misc.py]: "0-based indices"

File: qcelemental/util/misc.py [hunk 3]
```diff
@@ -130,3 +130,173 @@ def unnp(dicary, flat=False, _path=None):
...
+    Parameters
+    ----------
+    points1 : np.ndarray
+        The first list of points, can be 1D or 2D
+    points2 : np.ndarray
+        The second list of points, can be 1D or 2D
+    points3 : np.ndarray
+        The third list of points, can be 1D or 2D
+    degrees : bool, options
+        Returns the angle in degress rather than radians if True
```
Comment (loriab) [inline qcelemental/util/misc.py]: "degrees" (ditto in dihedral)

File: qcelemental/models/molecule.py [hunk 4]
```diff
@@ -228,6 +228,13 @@ def json(self, *args, **kwargs):
 
 ### Non-Pydantic API functions
 
+    def measure(self, measurements, degrees=True):
```
Comment (Lnaden) [inline qcelemental/models/molecule.py]: Are degrees really the default people expect for quantum chemistry applications? Just a side thought is all

File: qcelemental/util/misc.py (hunk 1 above)
Comment (dgasmith) [reply qcelemental/util/misc.py]: No, this is just a fast norm equivalent to `np.linalg.norm(arr, axis=1)`. Somewhat meaningless, but something I often do.

File: qcelemental/models/molecule.py (hunk 4 above)
Comment (dgasmith) [reply qcelemental/models/molecule.py]: I think so, happy to take other comments here however,

Comment (loriab) [reply qcelemental/models/molecule.py]: user-facing is usually degrees, I think. coders would rather convert than deal with confused gen chem students running labs and touting vespr. :-)

**Pull request merged**